app/helpers/administrate/application_helper.rb

```diff
@@ -38,10 +38,10 @@ def dashboard_from_resource(resource_name)
       "#{resource_name.to_s.singularize}_dashboard".classify.constantize
     end
 
-    def display_resource_name(resource_name)
+    def display_resource_name(resource_name, plural:)
```

Comment:
This won't be API compatible so when writing CHANGELOG it needs to be noted. Should pull requests already make entries?

Reply:
Is there a reasonable way we could make it backwards compatible? Or would that cause more problems than it solves?

Reply:
The naming is unexpected at the moment as the return value is plural but indicated in the signature so in my opinion it needs to go eventually. Deprecation can start with just a warning but the error message procuded to the call site with my addition should be clear enough.

Reply:
I think that's a good strategy. Do you think you'd be able to add something using [`ActiveSupport::Deprecation`](https://api.rubyonrails.org/classes/ActiveSupport/Deprecation.html)

Reply:
Probably